xds/internal/internal.go

```diff
@@ -37,20 +39,21 @@ type LocalityID struct {
 // String generates a string representation of LocalityID by adding ":" between
 // the components of the LocalityID.
 func (l LocalityID) String() string {
-	return fmt.Sprintf("%s:%s:%s", l.Region, l.Zone, l.SubZone)
+	b, err := json.Marshal(l)
+	if err != nil {
+		grpclog.Errorf("failed to marshal LocalityID: %#v", l)
```

Comment:
Vet is unhappy about this.

Reply:
Made it happy...

Reply:
That workaround seems against the spirit of the vet check, doesn't it?  Return the error instead or pass in the component logger (and don't use `String` as the method name in either case)?

Reply:
Done